hibernate-core/src/main/java/org/hibernate/loader/ast/internal/LoaderSelectBuilder.java

```diff
@@ -495,7 +508,7 @@ else if ( fetchDepth > maximumFetchDepth ) {
 			}
 
 			try {
-				if ( !( fetchable instanceof BasicValuedModelPart ) ) {
+				if ( !( fetchable instanceof BasicValuedModelPart ) && !(fetchable instanceof EmbeddedAttributeMapping ) ) {
```

Comment:
I am a little bit shamed to point out adding another space char after `(` in the above new code would make the format more consistent. Super minor for sure.